src/t2_rv32_template.c

```diff
@@ -0,0 +1,2143 @@
+
+RVT2OP(nop, { return; })
+
+RVT2OP(lui, {
+    LLVMValueRef rd_offset[1] = {LLVMConstInt(
```

Comment:
Could you provide some helper macros to simplify the invocation of LLVM APIs? This could reduce the effort required when we need to upgrade LLVM versions.